cda-main/src/main.rs

```diff
@@ -61,7 +63,7 @@ struct AppArgs {
 
 #[tokio::main]
 #[tracing::instrument]
-async fn main() -> Result<(), String> {
+async fn main() -> Result<(), DiagServiceError> {
```

Comment:
todo: I don't think that DiagServiceError is the correct error to use there. Probably one of my suggestion lead to this error ending up here. 
It probably would be best to introduce something like 'AppError' which then can be returned. 
DiagServiceError should only be used for diagnostic services and related issues. 
Therefore a failure to setup tracing is also not a DiagServiceError, as no diagnostic service was involved here.

I apologize if I caused this confusion.